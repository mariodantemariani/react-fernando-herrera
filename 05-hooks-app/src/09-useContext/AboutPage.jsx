export const AboutPage = () => {
  return (
    <>
      <h1>About APp</h1>
      <hr />
    </>
  );
};
