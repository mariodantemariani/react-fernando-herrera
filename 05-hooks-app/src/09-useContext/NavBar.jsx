import { Link, NavLink } from "react-router-dom";

export const NavBar = () => {
  return (
    <>
      <nav className="navbar navbar-expand-lg navbar-dark bg-dark rounded-3">
        <div className="container-fluid">
          <Link className="navbar-brand" href="#">
            usContext
          </Link>
          <div className="collapse navbar-collapse" id="navbarNav">
            <ul className="navbar-nav">
              <NavLink
                className={(args) =>
                  `nav-link ${args.isActive ? "active" : ""}`
                }
                to="/"
              >
                Home
              </NavLink>
              {/* <li className="nav-item">
                <Link className="nav-link" href="#">
                  Home
                </Link>
              </li> */}
              <NavLink
                className={(args) =>
                  `nav-link ${args.isActive ? "active" : ""}`
                }
                to="/about"
              >
                About
              </NavLink>

              <NavLink
                className={(args) =>
                  `nav-link ${args.isActive ? "active" : ""}`
                }
                to="/login"
              >
                Login
              </NavLink>

              {/* <li className="nav-item ">
                <Link className="nav-link active" href="#">
                  Features
                </Link>
              </li> */}
            </ul>
          </div>
        </div>
      </nav>
    </>
  );
};
