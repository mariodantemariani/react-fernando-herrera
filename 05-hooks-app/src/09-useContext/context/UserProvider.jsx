import { UserContext } from "./UserContext";
import { useState } from "react";
import PropTypes from "prop-types";

// const user = {
//   name: "Fernando",
//   id: 123,
//   email: "f@f.com",
// };
export const UserProvider = ({ children }) => {
  const [user, setUser] = useState();

  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};

UserProvider.propTypes = {
  children: PropTypes.array,
};
