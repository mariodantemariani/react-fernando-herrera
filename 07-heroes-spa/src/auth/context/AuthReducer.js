import { Types } from "../types/types";

// const initialState = {
//   logged: false,
//   name: "Mario",
// };

export const authReducer = (state = {}, action) => {
  switch (action.type) {
    case Types.login:
      return {
        ...state,
        logged: true,
        user: action.payload,
      };

    case Types.logout:
      return {
        logged: false,
      };

    default:
      return state;
  }
};
