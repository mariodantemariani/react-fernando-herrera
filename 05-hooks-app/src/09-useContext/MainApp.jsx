import { Navigate, Route, Routes } from "react-router-dom";

import { AboutPage, LoginPage, HomePage } from ".";

import { NavBar } from "../09-useContext/NavBar";
import { UserProvider } from "../09-useContext/context/UserProvider";

export const MainApp = () => {
  return (
    <UserProvider>
      {/* <h1>Main APP</h1> */}
      <NavBar />
      <hr />
      <Routes>
        <Route path="/" element={<HomePage />} />
        <Route path="about" element={<AboutPage />} />
        <Route path="login" element={<LoginPage />} />
        {/* <Route path="*" element={<HomePage />} /> */}
        <Route path="/*" element={<Navigate to="/about" />} />
      </Routes>
    </UserProvider>
  );
};
