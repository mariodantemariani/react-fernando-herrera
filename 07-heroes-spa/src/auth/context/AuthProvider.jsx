import { useReducer } from "react";
import { AuthContext } from "./AuthContext";
import PropTypes from "prop-types";
import { authReducer } from "./AuthReducer";
import { Types } from "../types/types";

// const initialState = {
//   logged: false,
//   name: "user",
// };

const init = () => {
  const user = JSON.parse(localStorage.getItem("user"));
  return { logged: !!user, user };
};

export const AuthProvider = ({ children }) => {
  const [authState, dispatch] = useReducer(authReducer, {}, init);

  const login = (name = "") => {
    const user = { id: 123, name };

    const action = {
      type: Types.login,
      payload: user,
    };

    localStorage.setItem("user", JSON.stringify(user));
    dispatch(action);
  };

  const logout = () => {
    localStorage.removeItem("user");
    const action = {
      type: Types.logout,
    };
    dispatch(action);
  };

  return (
    <AuthContext.Provider
      value={{
        ...authState,
        login,
        logout,
      }}
    >
      {children}
    </AuthContext.Provider>
  );
};

AuthProvider.propTypes = {
  children: PropTypes.object,
};
