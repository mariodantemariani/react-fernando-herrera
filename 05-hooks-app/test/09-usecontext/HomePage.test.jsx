import { render, screen } from "@testing-library/react";
import { UserContext } from "../../src/09-useContext/context/UserContext";

import { HomePage } from "../../src/09-useContext/HomePage";

describe("Pruebas en HomePage", () => {
  const user = {
    id: 1,
    name: "Fernando",
  };

  test("should be the same that snapShot", () => {
    const { container } = render(
      <UserContext.Provider value={{ user: null }}>
        <HomePage />
      </UserContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });

  test("debe de mostrar el componente sin el usuario", () => {
    render(
      <UserContext.Provider value={{ user: null }}>
        <HomePage />
      </UserContext.Provider>
    );

    const preTag = screen.getByLabelText("pre"); // aria-label
    expect(preTag.innerHTML).toBe("null");
    // screen.debug()
  });

  test("debe de mostrar el componente con el usuario", () => {
    render(
      <UserContext.Provider value={{ user }}>
        <HomePage />
      </UserContext.Provider>
    );

    const preTag = screen.getByLabelText("pre"); // aria-label
    expect(preTag.innerHTML).toContain(JSON.stringify(user, null, 3));
    expect(preTag.innerHTML).toContain(user.name);
    expect(preTag.innerHTML).toContain(user.id.toString());
    //screen.debug();
  });
});
