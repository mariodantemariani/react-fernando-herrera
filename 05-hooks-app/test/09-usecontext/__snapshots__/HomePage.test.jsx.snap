// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en HomePage should be the same that snapShot 1`] = `
<div>
  <h1>
    Home APP 
  </h1>
  <hr />
  <pre
    aria-label="pre"
  >
    null
  </pre>
</div>
`;
